### PR TITLE
Edwin - Removed option for Admins to reset password for Owners

### PR DESCRIPTION
**Pull request by EdwinLauM**
# Description

Screen Shot 2023-05-22 at 7 42 14 PM

## Mainly changes explained:
    1. Created a function to check permissions on owner user accounts to avoid duplicate code. 
    2. Added a conditional to check if the current user role has the correct permissions on owner user accounts. If it does, render the "reset password" button, if not don't. 
 
## How to test:
    1. check into current branch
    2. do `npm run start:local` to run this PR locally
    4. Login an admin account
    5. go to Other Links -> User Management
    6. Check that there is no "reset password" button for all owner accounts. 

### Before:

Admin Account: 
Screen Shot 2023-05-22 at 7 49 12 PM

### After:

Admin Account:
Screen Shot 2023-05-22 at 7 47 40 PM

Owner Account: 
Screen Shot 2023-05-23 at 10 49 28 AM

### CODE REVIEW

**Comment by lchua2314** [comment]
I do not have an owner account, so I will ask: Did you test an owner account? I don't know if devs are allowed owner accounts and I don't know how to make one.

File: src/components/UserManagement/UserTableData.jsx [hunk 1]
```diff
@@ -24,6 +24,14 @@ const UserTableData = React.memo(props => {
     onReset(false);
   }, [props.isActive, props.resetLoading]);
 
+  const checkPermissionsOnOwner = () => {
+    return (props.user.role === 'Owner' && !hasPermission(
+      props.role,
+      'addDeleteEditOwners',
+      props.roles,
+      props.userPermissions) ? true : false
```
Comment (lchua2314) [inline src/components/UserManagement/UserTableData.jsx:32]: Just call "hasPermission" function and remove the ternary operators. hasPersmission() will return the correct boolean. Keep the "!" operator though.

**Comment by Cirakuze** [comment]
Confirming the expected behavior logged in as an admin:

| BEFORE | AFTER |
|--|--|
| #847 BEFORE | #847 AFTER |

**Review by lchua2314 — CHANGES_REQUESTED** [review]
Make the small change. Everything else looks good. Tested the feature and got the desired output:

Before:
Before

After:
After

**Comment by Cirakuze** [comment]
It seems only the button inside the cell is hidden, and it leaves a a column of empty cells. Have you explored hiding the entire column, rather than just the buttons inside the cells?

**Review by eduhorta — APPROVED** [review]
Working as intended. Nice job! 👍
image

**Review by NidaZaki — APPROVED** [review]
The changes look good. Good job at hiding the entire cell for the "Owner".
Before:
PR#847 - BEFORE

After:
PR#847

**Review by lchua2314 — DISMISSED** [review]
Nice job on the changes!

**Review by 20chen-7 — APPROVED** [review]
The function is intended.
Screenshot 2023-05-23 at 11 40 55 AM
Screenshot 2023-05-23 at 11 40 22 AM

**Comment by EdwinLauM** [comment]
> @EdwinLauM I do not have an owner account, so I will ask: Did you test an owner account? I don't know if devs are allowed owner accounts and I don't know how to make one. It looks like it'll work though.

I have tested it in an owner account. Edited the description to include a screenshot of how it looks in an owner user account

**Comment by 20chen-7** [comment]
> > @EdwinLauM I do not have an owner account, so I will ask: Did you test an owner account? I don't know if devs are allowed owner accounts and I don't know how to make one. It looks like it'll work though.
> 
> I have tested it in an owner account. Edited the description to include a screenshot of how it looks in an owner user account

**Comment by susannagy** [comment]
Confirming that when I logged in as Admin user, in your branch that the column is not visible:
UI-manage-users

**Review by susannagy — DISMISSED** [review]
👍

**Comment by one-community** [comment]
Column should only be invisible IF filtering by "Owner", otherwise the password functions will be lost for other classes... but there will always only be one (maybe two) Owners, so hiding the column for that user class isn't something I care about.

File: src/components/UserManagement/UserTableHeader.jsx [hunk 2]
```diff
@@ -47,7 +47,12 @@ const UserTableHeader = React.memo(props => {
       <th scope="col" id="usermanagement_resume_date">
         End Date
       </th>
-      <th scope="col" id="usermanagement_delete"></th>
+      {((
+        props.authRole !=="Owner" && 
+        props.roleSearchText !=="Owner") || 
+        props.authRole =="Owner") && (
```
Comment (Cirakuze) [inline src/components/UserManagement/UserTableHeader.jsx:53]: Seems this is the same condition as in `UserTableSearchHeader` lines from 56 to 58.
How about moving this into a helper function that'll be imported into both files and receives the thee values as params?
`(authRole, roleSearchText, authRole) => ((authRole !== "Owner" && roleSearchText !== "Owner") || authRole === "Owner");`

Comment (EdwinLauM) [reply src/components/UserManagement/UserTableHeader.jsx]: Moved the same condition from`UserTableHeader` and `UserTableSearchHeader` into a helper function and imported in both files.

**Pull request merged**